core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -4548,88 +4549,95 @@ protected SqlBuilder getQueryString(boolean isGroupFiltering, boolean isUsername
         SqlBuilder sqlBuilder;
         boolean hitGroupFilter = false;
         boolean hitClaimFilter = false;
-        int groupFilterCount = 0;
-        int claimFilterCount = 0;
+        List<SqlBuilder> mysqlSubSqlBuilders = new ArrayList<>();
+        boolean isGroupFilteringWithNEOperator = isGroupFilteringWithNotEqualOperator(isGroupFiltering, expressionConditions);
+        String groupUserJoin =  isGroupFilteringWithNEOperator ? " RIGHT JOIN " : " INNER JOIN ";
 
         if (isGroupFiltering && isUsernameFiltering && isClaimFiltering || isGroupFiltering && isClaimFiltering) {
```

Comment:
The if conditions are not optimized (The issue was there before this PR.) Let's update the if conditions.
ex:         
if (isGroupFiltering && isClaimFiltering) {

}

Reply:
Also let's move this condition to inside isGroupFiltering matching conditions instead of a variable defined as main local variable of this method.

boolean isFilteringWithNEOperator = isGroupFilteringWithNotEqualOperator(expressionConditions);

Reply:
> The if conditions are not optimized (The issue was there before this PR.) Let's update the if conditions.
> ex:
> if (isGroupFiltering && isClaimFiltering) {
> 
> }

IMO there is no point of having a separate if condition because regardless username filtering requirement we still have to merge the `UM_USER` , which finally would be the same as the queries in the `if (isGroupFiltering && isUsernameFiltering && isClaimFiltering || isGroupFiltering && isClaimFiltering) {`

Reply:
Let's refactor the if conditions as suggested.
Also we can have `isGroupFilteringWithNEOperator` variable defined within if blocks.

Reply:
The variable is used in both of the following conditions:
`isGroupFiltering && isClaimFiltering`
and
`isGroupFiltering && isUsernameFiltering || isGroupFiltering`.

Since it’s needed in both, IMO we can't move the `isGroupFilteringWithNEOperator` and `groupUserJoin ` declaration inside the if block.

Reply:
I mean moving to both if blocks